src/page/mod.rs

```diff
@@ -63,6 +63,27 @@ pub enum CacheEntry<M> {
 
 unsafe impl<M: Send> Send for CacheEntry<M> {}
 
+#[derive(Debug, Clone)]
+pub struct CasKey<P> {
+    ptr: *const stack::Node<CacheEntry<P>>,
+    tag: usize,
+}
```

Comment:
I've created this to act like a sort of "access key" that lets other threads later try to atomically update shared state with, but without having access to the pointer underneath. This lets us avoid using coco `Scope`'s in higher-level API's above the pagecache, while still having the advantages of epoch GC. I'll write more about this in a trip report for adapting to coco, since it's sort of a side-effect of the way scopes are handled there.